File: src/main/java/com/safer/RH/models/Poste.java
package com.safer.RH.models;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data
@Entity
@RequiredArgsConstructor
public class Poste {
    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.IDENTITY)
    private int id;

    private String reference;
    @Column(nullable = false, unique = true)
    private String libelle;
    private String niveauEtude;
    private String description;
    private double niveauDeSalaire;
    private String remarque;
    @ManyToOne()
    private Secteur secteur;



}

File: src/app/api/models/Poste.ts
import { Secteur } from "./secteur";

export interface Poste{
    id?:number;
    reference?:string;
    libelle?:string;
    niveauEtude?:string;
    description?:string;
    niveauDeSalaire?:number;
    remarque?:string;
    secteur?:Secteur;
}